### PR TITLE
holdings: fix deletion of standard holdings

### PR DESCRIPTION
**Pull request by lauren-d**
Sometimes an holding record is already deleted from the database, 
but the index is not up to date. In this case, we need to delete it from index.

* Fixes `delete_standard_holdings_having_no_items` task.

Co-Authored-by: Lauren-D <laurent.dubois@itld-solutions.be>

### CODE REVIEW

File: rero_ils/modules/holdings/tasks.py [hunk 1]
```diff
@@ -36,13 +36,19 @@ def delete_standard_holdings_having_no_items():
...
+                reasons = record.reasons_not_to_delete()
+                current_app.logger.error(
+                    f'Can not delete standard holding: '
+                    f'{hit.pid} {reasons} {err}')
+        else:
+            # delete holding from index
+            HoldingsIndexer().client.delete(
+                id=hit.meta.id, index='holdings', doc_type='_doc')
 
     counts = {'count': es_query.count(), 'errors': errors}
```
Comment (rerowep) [inline rero_ils/modules/holdings/tasks.py:53]: May be the count is not correct here. Is this the remaining count or the count we tried to delete?

File: rero_ils/modules/holdings/tasks.py [hunk 2]
```diff
@@ -36,13 +36,19 @@ def delete_standard_holdings_having_no_items():
...
     for hit in [hit for hit in es_query.scan()]:
         record = Holding.get_record(hit.meta.id)
-        try:
-            record.delete(force=False, dbcommit=True, delindex=True)
-        except Exception as err:
-            errors += 1
-            reasons = record.reasons_not_to_delete()
-            current_app.logger.error(
-                f'Can not delete standard holding: {hit.pid} {reasons} {err}')
+        if record:
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/holdings/tasks.py]: Done

File: rero_ils/modules/holdings/tasks.py [hunk 3]
```diff
@@ -36,13 +36,19 @@ def delete_standard_holdings_having_no_items():
...
+                record.delete(force=False, dbcommit=True, delindex=True)
+            except Exception as err:
+                errors += 1
+                reasons = record.reasons_not_to_delete()
+                current_app.logger.error(
+                    f'Can not delete standard holding: '
+                    f'{hit.pid} {reasons} {err}')
+        else:
+            # delete holding from index
+            HoldingsIndexer().client.delete(
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/holdings/tasks.py:54]: Yes, but it seems less efficient.
Does it do two http requests : one for search and other one to delete?

File: rero_ils/modules/holdings/tasks.py (hunk 1 above)
Comment (lauren-d) [reply rero_ils/modules/holdings/tasks.py]: I've rewrite that

File: rero_ils/modules/holdings/tasks.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/holdings/tasks.py:54]: Yes of course. 
In my opinion , delete by query seems less efficient.
this delete all documents of a search results.
But may be I'm wrong.

**Review by zannkukai — APPROVED** [review]

**Review by benerken — APPROVED** [review]

**Review by rerowep — APPROVED** [review]

**Pull request merged**